context/MqlContext/MqlContextProvider.tsx

```diff
@@ -64,79 +76,67 @@ function MqlContextProviderInternal({
     for MqlServerUrl.
   */
   const [
-    {
-      data: mqlServerUrlData,
-      error: mqlServerUrlError,
-      fetching: mqlServerUrlfetching,
-    },
+    { data: mqlServerUrlData, error: mqlServerUrlError },
     refetchMqlServerUrl,
-  ] = useQuery<MqlServerUrlQuery>({
-    query: gql`
-      query MqlServerUrlQuery {
-        mqlServerUrl
-      }
-    `,
+  ] = useQuery<MqlServerUrlQueryType>({
+    query: MqlServerUrlQuery,
+    pause: !isAuthenticated,
   });
-  if (mqlServerUrlError) throwCarefully(mqlServerUrlError);
+  if (mqlServerUrlError) handleCombinedError(mqlServerUrlError);
 
   const [{ fetching }, setMqlServerUrl] = useMutation<
-    SetMqlServerUrl,
-    SetMqlServerUrlVariables
-  >(
-    gql`
-      mutation SetMqlServerUrl($newServerIdAsString: String!) {
-        createUserPreference(
-          object: {
-            prefKey: "mql_server_override_id"
-            prefValue: $newServerIdAsString
-          }
-          on_conflict: {
-            constraint: user_prefs_user_id_pref_key_key
-            update_columns: prefValue
-          }
-        ) {
-          id
-          prefKey
-          prefValue
-          userId
-          createdAt
-        }
-      }
-    `
-  );
+    SetMqlServerUrlMutationType,
```

Comment:
Tiny nit: This naming convention is a little confusing.  This method is going to be exposed and I think will lead to people wanting to supply the actual URL rather than the id for the mqlServer.  Also, where do they get the ID from?  Do they have some easy way to know which URL is which Id?

Reply:
Yes that makes sense. I'll update.

As for where the ID comes from, this mutation is currently used by our app on the MQL Server list page, which means the user has access to IDs for every row in the table. I could include a sample query for MQL Server ID and name to be instructive, even if we don't use that query ourselves.

Reply:
I think providing the method would be a nice addition.  Maybe a fast follow?  I believe slack was the reason we build the MQL Server list page so they must want the ability to easily switch servers?